typescript/agentkit/src/wallet-providers/cdpV2EvmWalletProvider.ts

```diff
@@ -0,0 +1,272 @@
+      apiKeyId,
+      apiKeySecret,
+      walletSecret,
+    });
+
+    const serverAccount = await (config.address
+      ? cdpClient.evm.getAccount({ address: config.address as Address })
+      : cdpClient.evm.createAccount({ idempotencyKey }));
+
+    const publicClient = createPublicClient({
+      chain: NETWORK_ID_TO_VIEM_CHAIN[networkId],
+      transport: http(),
+    });
+
+    return new CdpV2EvmWalletProvider({
+      publicClient,
+      cdpClient,
+      serverAccount,
+      network,
+    });
+  }
+
+  /**
+   * Signs a message.
+   *
+   * @param message - The message to sign.
+   * @returns The signed message.
+   */
+  async signMessage(message: string): Promise<Hex> {
+    return this.#serverAccount.signMessage({ message });
+  }
+
+  /**
+   * Signs a typed data object.
+   *
+   * @param typedData - The typed data object to sign.
+   * @returns The signed typed data object.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async signTypedData(typedData: any): Promise<Hex> {
+    return this.#serverAccount.signTypedData(typedData);
+  }
+
+  /**
+   * Signs a transaction.
+   *
+   * @param transaction - The transaction to sign.
+   * @returns The signed transaction.
+   */
+  async signTransaction(transaction: TransactionRequest): Promise<Hex> {
+    const serializedTx = serializeTransaction(transaction as TransactionSerializable);
+    const signedTx = await this.#cdpClient.evm.signTransaction({
+      address: this.#serverAccount.address,
+      transaction: serializedTx,
+    });
+
+    return signedTx.signature;
+  }
+
+  /**
+   * Sends a transaction.
+   *
+   * @param transaction - The transaction to send.
+   * @returns The hash of the transaction.
+   */
+  async sendTransaction(transaction: TransactionRequest): Promise<Hex> {
+    const result = await this.#cdpClient.evm.sendTransaction({
+      address: this.#serverAccount.address,
+      transaction: serializeTransaction(transaction as TransactionSerializable),
+      network: this.#getCdpSdkNetwork(),
+    });
+    return result.transactionHash;
+  }
+
+  /**
+   * Gets the address of the wallet.
+   *
+   * @returns The address of the wallet.
+   */
+  getAddress(): string {
+    return this.#serverAccount.address;
+  }
+
+  /**
+   * Gets the network of the wallet.
+   *
+   * @returns The network of the wallet.
+   */
+  getNetwork(): Network {
+    return this.#network;
+  }
+
+  /**
+   * Gets the name of the wallet provider.
+   *
+   * @returns The name of the wallet provider.
+   */
+  getName(): string {
+    return "cdp_v2_wallet_provider";
+  }
+
+  /**
+   * Gets the CDP client.
+   *
+   * @returns The CDP client.
+   */
+  getClient(): CdpClient {
+    return this.#cdpClient;
+  }
+
+  /**
+   * Gets the balance of the wallet.
+   *
+   * @returns The balance of the wallet in wei
+   */
+  async getBalance(): Promise<bigint> {
+    return await this.#publicClient!.getBalance({ address: this.#serverAccount.address });
```

Comment:
nit: it seems like `this.#publicClient` is definitely defined in the constructor, might not need to use `!` here